src/pages/SupplyRequisitionPage.js

```diff
@@ -0,0 +1,334 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import autobind from 'react-autobind';
+import { View } from 'react-native';
+
+import { GenericPage } from './GenericPage';
+import globalStyles from '../globalStyles';
+import { formatDate, parsePositiveInteger, sortDataBy } from '../utilities';
+import { buttonStrings, modalStrings, pageInfoStrings, tableStrings } from '../localization';
+import {
+  PageButton,
+  PageInfo,
+  PageContentModal,
+  TextEditor,
+} from '../widgets';
+
+const DATA_TYPES_SYNCHRONISED = ['RequisitionItem', 'Item', 'ItemBatch'];
+const MODAL_KEYS = {
+  COMMENT_EDIT: 'commentEdit',
+  ITEM_SELECT: 'itemSelect',
+  MONTHS_SELECT: 'monthsSelect',
+};
+
+export class SupplyRequisitionPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalKey: null,
+      modalIsOpen: false,
+      selection: [],
+    };
+    this.dataFilters = {
+      searchTerm: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    autobind(this);
+  }
+
+  /**
+   * Respond to the user editing the number in the required quantity column
+   * @param  {string} key             Should always be 'requiredQuantity'
+   * @param  {object} requisitionItem The requisition item from the row being edited
+   * @param  {string} newValue        The value the user entered in the cell
+   * @return {none}
+   */
+  onEndEditing(key, requisitionItem, newValue) {
+    // This will update associated CustomerInvoice if one exists or if linked
+    // CustomerInvoice does not exist, suppliedQuantity will not be updated
+    if (key !== 'suppliedQuantity') return;
+    const { database } = this.props;
+    database.write(() => {
+      const transactionItem = requisitionItem.linkedTransactionItem;
+      if (!transactionItem) return;
+      transactionItem.setTotalQuantity(database, parsePositiveInteger(newValue),
+                                                 transactionItem.availableQuantity);
+      requisitionItem.suppliedQuantity = transactionItem.totalQuantity;
+      database.save('TransactionItem', transactionItem);
+      database.save('RequisitionItem', requisitionItem);
```

Comment:
Basically the rule is:
At the point you use dot notation to change a data object, you must call .save
If you just call a method that manipulates the data object, you don't call .save. You can feel save in the knowledge that if the method uses dot notation to change it, it is the methods responsibility to call .save - so long as writer of the method followed the rules, you are fine (perhaps a good idea to check inside the method if you want to be really sure)